homeassistant/components/solarlog/config_flow.py

```diff
@@ -1,31 +1,39 @@
 """Config flow for solarlog integration."""
 
+from collections.abc import Mapping
 import logging
 from typing import TYPE_CHECKING, Any
 from urllib.parse import ParseResult, urlparse
 
 from solarlog_cli.solarlog_connector import SolarLogConnector
-from solarlog_cli.solarlog_exceptions import SolarLogConnectionError, SolarLogError
+from solarlog_cli.solarlog_exceptions import (
+    SolarLogAuthenticationError,
+    SolarLogConnectionError,
+    SolarLogError,
+)
 import voluptuous as vol
 
 from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
-from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PASSWORD
 from homeassistant.util import slugify
 
-from .const import DEFAULT_HOST, DEFAULT_NAME, DOMAIN
+from . import SolarlogConfigEntry
+from .const import CONF_HAS_PWD, DEFAULT_HOST, DEFAULT_NAME, DOMAIN
 
 _LOGGER = logging.getLogger(__name__)
 
 
 class SolarLogConfigFlow(ConfigFlow, domain=DOMAIN):
     """Handle a config flow for solarlog."""
 
+    _entry: SolarlogConfigEntry | None = None
     VERSION = 1
-    MINOR_VERSION = 2
+    MINOR_VERSION = 3
 
     def __init__(self) -> None:
         """Initialize the config flow."""
         self._errors: dict = {}
+        self._user_input: dict = {}
```

Comment:
Sharing errors and user input is a bad pattern since it's specific to each step form and should not leak out to other steps.